pilot/pkg/status/state.go

```diff
@@ -270,6 +270,7 @@ func ReconcileStatuses(current *config.Config, desired Progress, generation int6
 		if c.Type == "Reconciled" {
 			currentCondition = currentStatus.Conditions[i]
 			conditionIndex = i
+			break
```

Comment:
This changes semantics if there are multiple "Reconciled". In general it would be great to focus on fixing issues impacting real users - cleaning up minor code issues is ok but introduces risk for little benefit.

Thanks for your contributions!